ponyc no longer builds on main.

### Steps To Reproduce
Steps to reproduce the behavior:
1. build ponyc (or, actually - don't)

### Build log

Can be seen here: https://hydra.nixos.org/build/208066391/nixlog/1 - but in summary:

```
/build/source/src/libponyrt/lang/socket.c:51:10: fatal error: 'linux/dn.h' file not found
#include <linux/dn.h>
         ^~~~~~~~~~~~
1 error generated.
make[3]: *** [src/libponyrt/CMakeFiles/libponyrt.dir/build.make:430: src/libponyrt/CMakeFiles/libponyrt.dir/lang/socket.c.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:393: src/libponyrt/CMakeFiles/libponyrt.dir/all] Error 2
```

### Additional context

The new kernel headers removed DECNet and libponyrt had a dependency on it.  The excellent @stefantalpalaru opened a PR against ponyc to remove that dependency https://github.com/ponylang/ponyc/pull/4286 which was merged an in the current release.

### Notify maintainers

OHHAI @redvers 

(Note: I will be raising a PR to fix this in the next few hours)

--- a/pkgs/development/compilers/ponyc/default.nix
+++ b/pkgs/development/compilers/ponyc/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub, fetchurl, makeWrapper, pcre2, coreutils, which, openssl, libxml2, cmake, z3, substituteAll, python3,
+{ lib, stdenv, fetchFromGitHub, fetchpatch, makeWrapper, pcre2, coreutils, which, openssl, libxml2, cmake, z3, substituteAll, python3,
   cc ? stdenv.cc, lto ? !stdenv.isDarwin }:
 
 stdenv.mkDerivation (rec {
@@ -35,6 +35,11 @@ stdenv.mkDerivation (rec {
         rev = "release-1.10.0";
         sha256 = "1zbmab9295scgg4z2vclgfgjchfjailjnvzc6f5x9jvlsdi3dpwz";
       };
+    })
+    (fetchpatch {
+      name = "remove-decnet-header.patch";
+      url = "https://github.com/ponylang/ponyc/commit/e5b9b5daec5b19415d519b09954cbd3cf5f34220.patch";
+      hash = "sha256-60cOhBBwQxWLwEx+svtFtJ7POQkHzJo2LDPRJ5L/bNk=";
     })
   ];
 
@@ -93,6 +98,6 @@ stdenv.mkDerivation (rec {
     homepage = "https://www.ponylang.org";
     license = licenses.bsd2;
     maintainers = with maintainers; [ kamilchm patternspandemic redvers ];
-    platforms = [ "x86_64-linux" "x86_64-darwin" ];
+    platforms = [ "x86_64-linux" "x86_64-darwin" "aarch64-linux" ];
   };
 })

